### PR TITLE
fix bug and overload method

### PR DESCRIPTION
**Pull request by Chalin-Shi**
1. 🐛 fix 2 array equals' bug;
2. 🎨 overload 3 methods.

### CODE REVIEW

File: src/enforcer.ts [hunk 1]
```diff
@@ -511,8 +511,9 @@ export class Enforcer {
   /**
    * addPolicy adds a rule to the current policy.
    */
-  public async addPolicy(sec: string | any[], key?: string, rule?: string[]): Promise<boolean> {
-    if (typeof sec === 'string' && key && rule) {
+  public async addPolicy(...params: any[]): Promise<boolean> {
```
Comment (hsluoyz) [inline src/enforcer.ts]: Why is this changed? The parameters become unclear now.

File: src/enforcer.ts [hunk 2]
```diff
@@ -527,18 +528,17 @@ export class Enforcer {
...
+      return await this.addNamedPolicy('p', ...params);
     }
   }
 
   /**
    * removePolicy removes a rule from the current policy.
    */
-  public async removePolicy(sec: string | any[], key?: string, rule?: string[]): Promise<boolean> {
-    if (typeof sec === 'string' && key && rule) {
+  public async removePolicy(...params: any[]): Promise<boolean> {
```
Comment (hsluoyz) [inline src/enforcer.ts]: Why is this changed? The parameters become unclear now.

File: src/enforcer.ts [hunk 3]
```diff
@@ -553,18 +553,18 @@ export class Enforcer {
...
     }
   }
 
   /**
    * removeFilteredPolicy removes rules based on field filters from the current policy.
    */
-  public async removeFilteredPolicy(sec: string | number, key: string | string[], fieldIndex?: number, fieldValues?: string[]): Promise<boolean> {
-    if (typeof sec === 'string' && typeof key === 'string' && fieldIndex && fieldValues instanceof Array) {
+  // public async removeFilteredPolicy(sec: string | number, key: string | string[], fieldIndex?: number, fieldValues?: string[]): Promise<boolean> {
+  public async removeFilteredPolicy(...params: any[]): Promise<boolean> {
```
Comment (hsluoyz) [inline src/enforcer.ts]: Why is this changed? The parameters become unclear now.

File: src/enforcer.ts [hunk 4]
```diff
@@ -807,7 +808,7 @@ export class Enforcer {
    * @param params the "p" policy rule.
    * @return succeeds or not.
    */
-  public async addNamedPolicy(ptype: string, params: any[]): Promise<boolean> {
+  public async addNamedPolicy(ptype: string, ...params: any[]): Promise<boolean> {
```
Comment (hsluoyz) [inline src/enforcer.ts]: Why changes this?

File: src/enforcer.ts [hunk 5]
```diff
@@ -825,7 +826,7 @@ export class Enforcer {
    * @param params the "p" policy rule.
    * @return succeeds or not.
    */
-  public async removeNamedPolicy(ptype: string, params: any[]): Promise<boolean> {
+  public async removeNamedPolicy(ptype: string, ...params: any[]): Promise<boolean> {
```
Comment (hsluoyz) [inline src/enforcer.ts]: Why changes this?

File: src/enforcer.ts (hunk 3 above)
Comment (Chalin-Shi) [reply src/enforcer.ts]: Like go APIs: 
In Public AddPolicy method, overload private addPolicy method.
